src/vis/bar/utils.ts

```diff
@@ -226,17 +236,19 @@ async function setPlotsWithGroups(
   const uniqueColVals: string[] = [...new Set(catColValues.resolvedValues.map((v) => v.val))] as string[];
 
   uniqueGroupVals.forEach((uniqueVal) => {
+    const allGroupObjsIds = new Set((groupColumn.resolvedValues as VisCategoricalValue[]).filter((c) => c.val === uniqueVal).map((c) => c.id));
     const finalAggregateValues = uniqueColVals
       .map((v) => {
         const allObjs = (catColValues.resolvedValues as VisCategoricalValue[]).filter((c) => c.val === v);
-        const allGroupObjs = (groupColumn.resolvedValues as VisCategoricalValue[]).filter((c) => c.val === uniqueVal);
-        const joinedObjs = allObjs.filter((allVal) => allGroupObjs.find((groupVal) => groupVal.id === allVal.id));
+        const allObjsIds = new Set(allObjs.map((o) => o.id));
+
+        const joinedObjs = allObjs.filter((allVal) => allGroupObjsIds.has(allVal.id));
```

Comment:
This would work, but unfortunately lodash's intersection doesn't have a function I can pass in to determine equality of an object. So since i'm technically comparing an array of objects to just an array of strings, it wouldn't work. And if I converted the array of objects to just its string ids first for the comparison, I would lose the other parts of the object which I need after the intersection

Reply:
I see. Yes, then we just leave it as it is.